packages/api/src/auth/decoders/dbAuth.ts

```diff
@@ -0,0 +1,17 @@
+import type { APIGatewayProxyEvent } from 'aws-lambda'
+
+import { dbAuthSession } from '../../functions/dbAuth/shared'
+
+export const dbAuth = (
+  authHeaderValue: string,
+  req: { event: APIGatewayProxyEvent }
+) => {
+  const session = dbAuthSession(req.event)
+  const authHeaderUserId = authHeaderValue
+
+  if (session.id.toString() !== authHeaderUserId) {
```

Comment:
the `toString()` part is weird

Reply:
I assume TS, just a comment. Not an actionable thing.

Reply:
SO that's because HTTP headers always come in as strings. But, the user's ID you get from the database could be a string (like a UUID) or could be an integer, depending on how it's stored. So this just makes sure it's always a string so that it can compare against he header.

Reply:
And I don't know that this check is even required (since the session is encrypted, it's extremely unlikely that the wrong user ID is in there) but I figured since *something* is gonna go up in that auth header, we may as well double check that it's the correct value.

Reply:
And that's just what I decided to put in that header, it could be anything, right? Right now it's the plain text of the user's ID, which some might consider a security risk (if it's an integer then you know you can start going through each of them in a row to try and hack something).